### PR TITLE
Fix error handling under cmd/root

### LINKED ISSUE #48: [BUG] error handling needs to be done correctly
**Describe the bug**
Currently, all errors that might occur are being returned up the function call stack with the intention of being returned to the files in the `cmd` directory. `cobra` provides functionality for handling errors by providing `Run` or `RunE` in the `cmdNameCmd` struct which allows you to return an error to `cobra` and it tries and handles it. 

With the current concurrency model of `grofer`, the functions for retrieving data and rendering data are put in go-routines because of why "returning" an error to that function is rendered pointless.

**Proposal:**
Have a channel of type `error` which blocks waiting for an event to occur, on a successful run, `nil` could be passed to the channel in which case the program can exit, whereas, on the occurrence of an error, the error can be taken out of the channel and handled or returned to another function if need be for handling. 

**To Reproduce**
1. Go to `src/general/cpuInfo.go`
2. Change "mpstat" to "baby-shark-dodododo" (hopefully not a valid command although that would be awesome) on line 52
3. Execute: `go run grofer.go --cpuinfo`
4. Everything renders fine.

**Expected behavior**
Error messages must be displayed. 

**Desktop (please complete the following information):**
 - OS: Kubuntu 20.04
 - Go Version: 1.14.7

**Additional context**
Since such errors aren't handled, the `kill` signal does not execute properly on pressing the `q` key which is meant to exit the program. There is an error that's occurred but that goroutine cannot handle it which is why the `kill` signal works in the rendering part but does not work correctly in the data serving part and it blocks there and the services of `ctrl + c` need to be employed.

### PR DESCRIPTION
**Pull request by egawata**
# Description

This PR fixes the way of error handling under `cmd/root`.

- When any error occurs, it will be returned to callers, which enables to handle it at the right place.
- Pressing `q` terminates execution in a right way.

Fixes #48 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# Checklist:

- [x] I have read the [contribution guidelines](https://github.com/pesos/grofer/blob/master/CONTRIBUTING.md) and followed it as far as possible. 
- [x] I have performed a self-review of my own code (if applicable)
- [x] I have commented my code, particularly in hard-to-understand areas (if applicable)
- [x] I have run `go fmt` on my code ([reference](https://blog.golang.org/gofmt))
- [x] I have made corresponding changes to the documentation (if applicable)
- [x] My changes generate no new warnings 
- [x] Any dependent and pending changes have been merged and published

### CODE REVIEW

File: src/general/errors.go [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package general
```
Comment (Gituser143) [inline src/general/errors.go:17]: Hi @egawata since this file is a new addition. We'll need you to put in the license at the top of the file.

**Review by Gituser143 — CHANGES_REQUESTED** [review]

File: src/general/errors.go (hunk 1 above)
Comment (MadhavJivrajani) [reply src/general/errors.go:17]: You can find the license at the beginning of any of the `.go` files

Comment (egawata) [reply src/general/errors.go:17]: @Gituser143 
I'm sorry for forgetting that 🙇‍♂️   I've added it at https://github.com/pesos/grofer/pull/53/commits/d48915bd6d30abf0a7aa2e7c2e73a50ddf01bd5c .

**Review by MadhavJivrajani — APPROVED** [review]
Great stuff @egawata ! Thanks for your contribution!

**Review by Gituser143 — APPROVED** [review]

**Pull request merged**